lib/crypto.js

```diff
@@ -518,9 +519,11 @@ function ECDH(curve) {
   this._handle = new binding.ECDH(curve);
 }
 
-exports.createECDH = function createECDH(curve) {
-  return new ECDH(curve);
-};
+if (hasEC()) {
+  exports.createECDH = function createECDH(curve) {
+    return new ECDH(curve);
+  };
+}
```

Comment:
@jasnell I've updated with two commits now. 

64b11e4 tries to address your suggestion of having a constant flag and removes the `hasEC` function. The constant, named `openssl_no_ec`, is check by `createECDH` and `getCurves` to throw an error.

1f55434 tries to reduce the amount of checks for OPENSSL_NO_EC but still allow compilation and tests to succeed.